flow/connectors/postgres/validate.go

```diff
@@ -139,6 +139,11 @@ func (c *PostgresConnector) CheckReplicationConnectivity(ctx context.Context) er
 	if err != nil {
 		return fmt.Errorf("failed to create replication connection: %v", err)
 	}
+	defer conn.Close(ctx)
```

Comment:
this will double close, move into error branch instead of defer

Reply:
pgx is fine with calling `.Close()` repeatedly

```
Close closes a connection. It is safe to call Close on an already closed connection.
```